internal/pkg/statemanager/state.go

```diff
@@ -0,0 +1,63 @@
+package statemanager
+
+import (
+	"bytes"
+	"log"
+
+	"gopkg.in/yaml.v3"
+)
+
+type CompStatus string
+type CompAction string
+
+const (
+	StatusUninstalled CompStatus = "uninstalled"
+	StatusInstalled   CompStatus = "installed"
+	StatusRunning     CompStatus = "running"
+	StatusFailed      CompStatus = "Failed"
+)
+
+const (
+	ActionInstall   CompAction = "install"
+	ActionReinstall CompAction = "reinstall"
+	ActionUninstall CompAction = "uninstall"
+)
+
+type States map[string]*State
+
+// State is the single component's state.
+type State struct {
```

Comment:
I suggest an extra field like "lastUpdatedTime", but it's not necessary at the moment.

Reply:
lastUpdatedTime may is duplicated with State.LastOperation.Time